pkg/action/hooks.go

```diff
@@ -28,6 +28,24 @@ import (
 
 // execHook executes all of the hooks for the given hook event.
 func (cfg *Configuration) execHook(rl *release.Release, hook release.HookEvent, timeout time.Duration) error {
+	shutdown, err := cfg.execHookWithDelayedShutdown(rl, hook, timeout)
+	if err != nil {
+		if err := shutdown(); err != nil {
```

Comment:
My IDE (GoLand)  warns for `shutdown()` in this line:
>  'shutdown' might have 'nil' or other unexpected value as its corresponding error variable might be not 'nil' 

From my implementation, I don't see how that could be `nil`. Since I am quite new to go: is there a possibility to specify that that function cannot be `nil`? Something like a `@Nonnull` in Java? Or would you ignore this warning or write these parts differently?

Reply:
for the type of the `shutdown` returned, the default value if none is assigned would be `nil`; uninitialized func variables are `nil`. Even though looking at the implementation for `execHookWithDelayedShutdown`, it does appear to "never" return a `nil` function, it's still possible from the perspective of the client, since the actual value of it cannot be inferred at compile time. To be safe (and to appease your IDE), you should also check for a `nil` shutdown function and use the no-op function if nil. 

That might mean that you can change your implementation to return a `nil` shutdown, and let users of your function determine what to do in such cases.

Reply:
Thanks @fcervantes2! I added 
````
        if shutdown == nil {
		return nil
	}
````
to the function. I checked for the usages of `execHook()` and they all do a nil check - so that should be indeed be handled already.